core/src/main/java/nordpol/Apdu.java

```diff
@@ -97,4 +101,72 @@ public static String encodeHex(byte b) {
         return Arrays.copyOfRange(response, 0, response.length - 2);
     }
 
+    /**
+     * Matches the first APDUs last characters (of the length of the second APDU) against the second APDU
+     * @param receivedApdu The response APDU as byte[] to be compared
+     * @param expectedApdu The expected APDU as byte[] to match against
+     * @return true if the APDUs match, false if not
+     */
+    static boolean requireStatus(byte[] receivedApdu, byte[] expectedApdu) throws IOException {
+        return requireStatus(encodeHex(receivedApdu), encodeHex(expectedApdu));
+    }
+
+    /**
+     * Matches the first APDUs last characters (of the length of the second APDU) against the second APDU
+     * @param receivedApdu The response APDU as byte[] to be compared
+     * @param expectedApdu The expected APDU as String to match against
+     * @return true if the APDUs match, false if not
+     */
+    static boolean requireStatus(byte[] receivedApdu, String expectedApdu) throws IOException {
+        return requireStatus(encodeHex(receivedApdu), expectedApdu);
+    }
+
+    /**
+     * Matches the first APDUs last characters (of the length of the second APDU) against the second APDU
+     * @param receivedApdu The response APDU as String to be compared
+     * @param expectedApdu The expected APDU as String to match against
+     * @return true if the APDUs match, false if not
+     */
+    static boolean requireStatus(String receivedApdu, String expectedApdu) throws IOException {
+        String concatReceivedApdu = shortenString(receivedApdu, expectedApdu.length());
+        return concatReceivedApdu.matches(expectedApdu);
```

Comment:
@Franzaine you should use http://docs.oracle.com/javase/7/docs/api/java/lang/String.html#equalsIgnoreCase%28java.lang.String%29 here